crates/core/src/oracle/constraint.rs

```diff
@@ -105,20 +104,6 @@ impl<F: Field> ConstraintSetBuilder<F> {
 		});
 	}
 
-	pub fn add_zerocheck(
-		&mut self,
-		name: impl ToString,
-		oracle_ids: impl IntoIterator<Item = OracleId>,
-		composition: ArithCircuit<F>,
-	) {
-		self.constraints.push(UngroupedConstraint {
-			name: name.to_string(),
-			oracle_ids: oracle_ids.into_iter().collect(),
-			composition,
-			predicate: ConstraintPredicate::Zero,
```

Comment:
If you plan to remove `add_zerocheck`, you will also need to refactor `ConstraintPredicate` and all usages of it.

Reply:
oh yeah I noticed that it's not used anywhere. It even panics in validate_witness it. You wan't me to do this in this PR?